backend/cmm_helpers.ml

```diff
@@ -4839,26 +4730,25 @@ let make_unboxed_int32_array_payload dbg unboxed_int32_list =
   in
   aux [] unboxed_int32_list
 
-let allocate_unboxed_int32_array ~elements (mode : Cmm.Alloc_mode.t) dbg =
-  let num_elts, payload = make_unboxed_int32_array_payload dbg elements in
+let allocate_unboxed_packed_array ~make_payload ~alloc_kind ~base_tag ~elements
+    mode dbg =
+  let num_elts, payload = make_payload dbg elements in
+  let tag = base_tag + match num_elts with Even -> 0 | Odd -> 1 in
   let header =
-    let size = 1 (* custom_ops field *) + List.length payload in
+    let size = List.length payload in
     match mode with
-    | Heap -> custom_header ~size
-    | Local -> custom_local_header ~size
-  in
-  let custom_ops =
-    (* For odd-length unboxed int32 arrays there are 32 bits spare at the end of
-       the block, which are never read. They are initialized to the sign
-       extension of the last element. *)
-    match num_elts with
-    | Even -> custom_ops_unboxed_int32_even_array
-    | Odd -> custom_ops_unboxed_int32_odd_array
+    | Cmm.Alloc_mode.Heap ->
+      black_mixed_block_header tag size ~scannable_prefix_len:0
```

Comment:
I have no idea what GC colour these should be, but this appears to be different to what it was.